File: data/recomposed_projects/commons-fileupload/src/main/org/apache/commons/fileupload/FileCountLimitExceededException.py
# Imports Begin
from src.main.org.apache.commons.fileupload.FileUploadException import *

from src.main.org.apache.commons.fileupload.java_handler import java_handler
# Imports End


@java_handler
class FileCountLimitExceededException(FileUploadException):

    # Class Fields Begin
    __serialVersionUID: int = 6904179610227521789
    __limit: int = None
    # Class Fields End

    # Class Methods Begin
    def getLimit(self) -> int:

        return self.__limit

    def __init__(self, message: str, limit: int) -> None:

        super().__init__(message, None)
        self.limit = limit

    # Class Methods End

File: java_projects/glue_code/commons-fileupload/src/main/java/org/apache/commons/fileupload/FileCountLimitExceededException.java
package org.apache.commons.fileupload;

import org.graalvm.polyglot.Value;

public class FileCountLimitExceededException extends FileUploadException {
  private static final long serialVersionUID = 6904179610227521789L;
  private static Value clz =
      ContextInitializer.getPythonClass(
          "FileCountLimitExceededException.py", "FileCountLimitExceededException");
  private Value obj;

  public FileCountLimitExceededException(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public long getLimit() {
    //
    // return limit;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("getLimit").as(long.class);
  }

  public FileCountLimitExceededException(final String message, final long limit) {
    //
    // super(message, null);
    // this.limit = limit;
    //

    this.obj = clz.invokeMember("__init__", message, limit);
  }
}
